src/components/Dashboard/index.tsx

```diff
@@ -173,6 +173,12 @@ export default class Dashboard extends React.Component<IDashboardProps, IDashboa
   }
 
   onDeleteDashboardApprove() {
+    let { dashboard } = this.props;
+    if (!dashboard) {
+      console.warn('Dashboard not found. Aborting delete.');
+    }
+    ConfigurationsActions.deleteDashboard(dashboard.id);
+    window.location.href = '/';
```

Comment:
This is a problem since you can't know if the deletion was successful or not.
Please move this to `ConfigurationsActions:deleteDashboard` ==> if success, redirect to '/'